sched/Kconfig

```diff
@@ -1000,6 +1000,26 @@ config SCHED_INSTRUMENTATION_FILTER
 		The filter logic can be configured by sched_note_filter APIs defined in
 		include/nuttx/sched_note.h.
 
+config SCHED_INSTRUMENTATION_FILTER_ENABLE_ONBOOT
```

Comment:
why need an additional config? it's simple to directly add SCHED_INSTRUMENTATION_FILTER_MODE and SCHED_INSTRUMENTATION_FILTER_ENABLE_ONBOOT_NOOVERWRITE.

Reply:
Because the filter mode contains the following variables:

1.  global instrumentation enable/disable
2.  syscall/irq instrumentation enable flag

SCHED_INSTRUMENTATION_FILTER_ENABLE_ONBOOT defines the default value of 1.
SCHED_INSTRUMENTATION_FILTER_ENABLE_ONBOOT_MODE defines 2. (The name may be confusing.)

@xiaoxiang781216 Do you mean that these two variables should be integrated to one mode variable and provide only one configuration which defines the default value of the mode variable?

Reply:
How about we change enable field to a flag part of flag field? so we can control the initial state by one Kconfig option.